disnake/ext/commands/interaction_bot_base.py

```diff
@@ -232,31 +252,131 @@ def command_sync_flags(self) -> CommandSyncFlags:
 
     @property
     def slash_commands(self) -> Set[InvokableSlashCommand]:
         """Set[:class:`InvokableSlashCommand`]: A set of all slash commands the bot has."""
-        return set(self.all_slash_commands.values())
+        return {
+            cmd for cmd in self.all_app_commands.values() if isinstance(cmd, InvokableSlashCommand)
+        }
 
     @property
     def user_commands(self) -> Set[InvokableUserCommand]:
         """Set[:class:`InvokableUserCommand`]: A set of all user commands the bot has."""
-        return set(self.all_user_commands.values())
+        return {
+            cmd for cmd in self.all_app_commands.values() if isinstance(cmd, InvokableUserCommand)
+        }
 
     @property
     def message_commands(self) -> Set[InvokableMessageCommand]:
         """Set[:class:`InvokableMessageCommand`]: A set of all message commands the bot has."""
-        return set(self.all_message_commands.values())
+        return {
+            cmd
+            for cmd in self.all_app_commands.values()
+            if isinstance(cmd, InvokableMessageCommand)
+        }
+
+    @property
+    def all_slash_commands(self) -> Dict[str, InvokableSlashCommand]:
+        # no docstring because it was an attribute and now it's deprecated
+        warn_deprecated(
+            "all_slash_commands is deprecated and will be removed in a future version. "
+            "Use all_app_commands as a replacement.",
+            stacklevel=3,
+        )
+        return {
+            cmd.name: cmd
+            for cmd in self.all_app_commands.values()
+            if isinstance(cmd, InvokableSlashCommand)
+        }
+
+    @property
+    def all_user_commands(self) -> Dict[str, InvokableUserCommand]:
+        # no docstring because it was an attribute and now it's deprecated
+        warn_deprecated(
+            "all_user_commands is deprecated and will be removed in a future version. "
+            "Use all_app_commands as a replacement.",
+            stacklevel=3,
+        )
+        return {
+            cmd.name: cmd
+            for cmd in self.all_app_commands.values()
+            if isinstance(cmd, InvokableUserCommand)
+        }
+
+    @property
+    def all_message_commands(self) -> Dict[str, InvokableMessageCommand]:
+        # no docstring because it was an attribute and now it's deprecated
+        warn_deprecated(
+            "all_message_commands is deprecated and will be removed in a future version. "
+            "Use all_app_commands as a replacement.",
+            stacklevel=3,
+        )
+        return {
+            cmd.name: cmd
+            for cmd in self.all_app_commands.values()
+            if isinstance(cmd, InvokableMessageCommand)
+        }
+
+    def add_app_command(self, app_command: InvokableApplicationCommand) -> None:
+        """Adds an :class:`InvokableApplicationCommand` into the internal list of app commands.
+
+        This is usually not called, instead shortcut decorators are used.
```

Comment:
Care to include a link to the decorators people should use?

Reply:
https://github.com/DisnakeDev/disnake/pull/1107/commits/e86e5cb646a2c3ba99c2bb0318acd3f2fec51572